segmentation/utils/synthetic_spatial_datagen_test.py

```diff
@@ -102,3 +106,70 @@ def test_generate_test_label_map():
+
+
+def test_generate_two_cell_test_segmentation_mask():
+    # this function tests the functionality of generating the segmentation mask
+
+    sample_segmentation_mask, sample_cell_centers = synthetic_spatial_datagen.generate_two_cell_test_segmentation_mask()
+
+    # assert that our labels are just blank, 1, and 2
+    assert set(np.unique(sample_segmentation_mask)) == set([0, 1, 2])
+
+    # assert that our centers are being labeled correctly
+    assert sample_segmentation_mask[sample_cell_centers[1][0], sample_cell_centers[1][0]] == 1
+    assert sample_segmentation_mask[sample_cell_centers[2][0], sample_cell_centers[2][1]] == 2
+
+    # using the default cell radius of 10, assert that the cells are next to each other
+    # we only include the offset columnwise because that's how the cells are generated next to each other
+    assert sample_segmentation_mask[sample_cell_centers[1][0], sample_cell_centers[1][1] + 10] == 2
+    assert sample_segmentation_mask[sample_cell_centers[2][0], sample_cell_centers[2][1] - 10] == 1
+
+
+def test_generate_two_cell_test_nuclear_signal():
+    # this function tests the functionality of the nuclear-signal-generating portion of
+    # the channel-level spatial analysis data
+
+    sample_segmentation_mask, sample_cell_centers = synthetic_spatial_datagen.generate_two_cell_test_segmentation_mask()
+
+    sample_nuclear_signal = synthetic_spatial_datagen.generate_two_cell_test_nuclear_signal(sample_segmentation_mask,
+                                                                                            sample_cell_centers)
+
+    # assert that our nucleus center is labeled properly
+    # we only care about cell 1 because that is the only nuclear-level expression cell by default
+    assert sample_nuclear_signal[sample_cell_centers[1][0], sample_cell_centers[1][1]] == 1
+
+    # because we'll be jittering the signal eventually, we won't test the status of the signal at the nucleus border
+
+
+def test_generate_two_cell_test_membrane_signal():
+    # this function tests the functionality of the membrane-signal-generating portion of
+    # the channel-level spatial analysis data
+
+    sample_segmentation_mask, sample_cell_centers = synthetic_spatial_datagen.generate_two_cell_test_segmentation_mask()
+
+    sample_membrane_signal = synthetic_spatial_datagen.generate_two_cell_test_membrane_signal(sample_segmentation_mask,
+                                                                                              sample_cell_centers)
+
+    # assuming the default membrane diameter of 5, assert that our membrane inner edge is being labeled correctly
+    # we only include the offset columnwise by choice: it could be done rowwise as well
+    # we only care about cell 2 because that is the only membrane-level expression cell by default
+    assert sample_membrane_signal[sample_cell_centers[2][0], sample_cell_centers[2][1] - 5] == 1
```

Comment:
see earlier comment on function default dependence (honestly, I think I might have to go back into some of my data_utils test functions to fix that kinda thing...)